File: cpp/src/LocationSensor.cpp
#include "LocationSensor.hpp"

LocationSensor::LocationSensor(const std::string &loc): currentLocation(loc) {}

LocationSensor::LocationSensor(const LocationSensor& other): currentLocation(other.currentLocation) {}

void LocationSensor::updateLocation(const std::string &loc) {
    currentLocation = loc;
    std::cout << "Датчик местоположения: местоположение обновлено на " << loc << std::endl;
}

std::string LocationSensor::getCurrentLocation() const {
    return currentLocation;
}

File: java/java/LocationSensor.java
package smarthome;

public class LocationSensor {

    private String currentLocation = "unknown";

    public void updateLocation(String loc) {
        currentLocation = loc;
        System.out.println("Локация обновлена: " + loc);
    }

    public void triggerLocationBasedScenarios(SmartHomeSystem system) {
        for (Scenario s : system.getScenarios()) {
            s.execute(system);
        }
    }

    public String getCurrentLocation() {
        return currentLocation;
    }

    public boolean isAtHome() {
        return "home".equalsIgnoreCase(currentLocation);
    }
}
